Add test so that bug 285568 can't happen again

jyavenard opened the pull request:
#### 4bdbf870a50a1150b3ecc5635072c0020f12c2d0
<pre>
Add test so that bug 285568 can&apos;t happen again
<a href="https://bugs.webkit.org/show_bug.cgi?id=285569">https://bugs.webkit.org/show_bug.cgi?id=285569</a>
<a href="https://rdar.apple.com/142517881">rdar://142517881</a>

Reviewed by Youenn Fablet.

287274@main broke media playback by not calling MediaPlayer::play() if we
were in the middle of a seek.
With 287274@main reverted in 288588@main, we replicate the code pattern used by duolingo to ensure
that such bug can&apos;t be re-introduced inadvertently in the future.

* LayoutTests/platform/glib/TestExpectations: ended event isn&apos;t fired the 2nd time. webkit.org/b/285595
* LayoutTests/media/content/short.wav: The file was broken. Replace its content with short.mp3 converted to wav using ffmpeg.
* LayoutTests/media/video-seek-after-end-play-expected.txt: Added.
* LayoutTests/media/video-seek-after-end-play.html: Added.

Canonical link: <a href="https://commits.webkit.org/288590@main">https://commits.webkit.org/288590@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b290d3bf1022042e8e66b0e62842d1d11ab8c8a1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/83878 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/3496 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/38179 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/88952 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/34886 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/3587 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/11463 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/65239 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/23077 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/86924 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/2657 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/76212 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/45531 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/2586 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/30433 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/33935 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/73580 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/31187 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/90328 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/11143 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/8053 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/73689 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/11367 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/72041 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/72908 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18023 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/17184 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/15872 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/2474 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/11095 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/10943 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/14419 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/12715 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf reviewed (approved):
Pre-existing but I wonder whether the second seeking should happen before the second play event.

jyavenard committed:
Add test so that bug 285568 can't happen again
https://bugs.webkit.org/show_bug.cgi?id=285569
rdar://142517881

Reviewed by Youenn Fablet.

287274@main broke media playback by not calling MediaPlayer::play() if we
were in the middle of a seek.
With 287274@main reverted in 288588@main, we replicate the code pattern used by duolingo to ensure
that such bug can't be re-introduced inadvertently in the future.

* LayoutTests/platform/glib/TestExpectations: ended event isn't fired the 2nd time. webkit.org/b/285595
* LayoutTests/media/content/short.wav: The file was broken. Replace its content with short.mp3 converted to wav using ffmpeg.
* LayoutTests/media/video-seek-after-end-play-expected.txt: Added.
* LayoutTests/media/video-seek-after-end-play.html: Added.

Canonical link: https://commits.webkit.org/288590@main
